src/utils/millions.ts

```diff
@@ -36,3 +40,20 @@ export const withdrawalStateToString = (state: WithdrawalState): string => {
             return 'UNKNOWN';
     }
 };
+
+export const groupAndSumDeposits = (deposits: Deposit[]): DepositWithRawAmount[] => {
+    const groupedDeposits: { [address: string]: DepositWithRawAmount } = {};
+
+    deposits.forEach((deposit) => {
```

Comment:
forEach is asynchronous operation. Make sure of operation orders. For loop should be better.